### ISSUE TITLE
problem parsing relative items in conf when combined with a function

### ISSUE THREAD

**Issue by onkelandy** [smarthomeNG/smarthome#184]
When using relative items in combination with a function, lets say a "changed_by()" or "last_update()" the relative item gets misinterpreted.

Easiest way to solve I guess is to search for () and then interpret the dots differently.
sh...changed_by() should look for the change of the item itself.. though having three consecutive points looks a bit weird ;)

**Comment by msinn** [comment]
I am looking into it

**Comment by msinn** [comment]
@onkelandy From what I found out, your problem only exists, if you want to address a function for the item itself (eg..: 'sh...changed_by()') and direct ancestors, but not for referencing other items like 'sh..sister.changed_by()'.

Interpreting the dots in another way if parentheses are present gives a lot of other problems. The solution I found is the following: Inserting a reserved word 'self'. 

To reference the item itself you can use '.self' instead of '.'. So instead of writing 'sh...changed_by()' (which produces unexpected results), you could write 'sh..self.changed_by()'.

Or for referencing a function of the parent item, instead of writing 'sh....changed_by()' (which produces unexpected results), you can write 'sh...self.changed_by()'

What do you think about that solution?

**Comment by onkelandy** [comment]
Sounds like a fair solution.. Though I think the dots in general are very irritating especially when combined with sh.
Wouldnt a delimiter, something like sh.[.].changed_by() work and make it more readable?

**Comment by msinn** [comment]
I am not sure, if it would make things even more complex. There has been a discussion about escape-chars for relative addressing in the past, but without a good solution.

**Comment by msinn** [comment]
It will take a couple of days for the modification to be included in the dev repository, because I am doing some renovation on the autotimer handling for items. Until then you could only use a workaround  with addressing using the parent item like:

```
current_item -> .
parent_item -> ..

sh...changed_by()                # is not working at the moment
sh...current_item.changed_by()   # workarount
```

**Comment by onkelandy** [comment]
thx! I've already changed the references to global ones but maybe it's useful for someone else.

**Comment by msinn** [comment]
Implemented for Version 1.3 and documented in wiki.
